### PR TITLE
Bump to polkadot-v0.9.22

### PR DESCRIPTION
**Pull request by yrong**
### upstream pr

https://github.com/paritytech/substrate/pull/8783
https://github.com/paritytech/substrate/pull/11241
https://github.com/paritytech/substrate/pull/11198

### checklist
- [x] bump to v0.9.22 & make code compile
- [x] introduce udeps & remove unused dependencies 
- [x]  fix breaking integration test
- [x]  forked relaychain customization & launch config change & some live test

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -11,4 +11,14 @@ lto = true
 codegen-units = 1
 
 [workspace]
-members = ['node/*', 'pallets/*', 'runtime/*', 'integration-tests','support']
\ No newline at end of file
+members = ['node/*', 'pallets/*', 'runtime/*','support','integration-tests']
+
+[patch.crates-io]
+orml-traits                 = { git = 'https://github.com/open-web3-stack/open-runtime-module-library', rev = 'f709ed62262435b3ad80482d309e3575625d1e5b' }
```
[the remark this answers is not in the document]
Comment (yrong) [reply Cargo.toml:17]: @GopherJ  could we relax check here?  not found 0.9.22 specific branch or tag in orml repo so we patch with commit id in this pr

File: integration-tests/Cargo.toml [hunk 2]
```diff
@@ -71,16 +71,16 @@ parallel-runtime              = { path = '../runtime/parallel' }
...
-xcm-emulator                          = { git = 'https://github.com/shaunxw/xcm-simulator.git', rev = '6c358483d8e119cd3b631ebb14d3b0cf0041d94e' }
+polkadot-primitives                   = { git = 'https://github.com/paritytech/polkadot.git', branch = 'release-v0.9.22' }
+polkadot-runtime                      = { git = 'https://github.com/paritytech/polkadot.git', branch = 'release-v0.9.22' }
+polkadot-runtime-parachains           = { git = 'https://github.com/paritytech/polkadot.git', branch = 'release-v0.9.22' }
+sp-io                                 = { git = 'https://github.com/paritytech/substrate.git', branch = 'polkadot-v0.9.22' }
+sp-trie                               = { git = 'https://github.com/paritytech/substrate.git', branch = 'polkadot-v0.9.22' }
+statemine-runtime                     = { git = 'https://github.com/paritytech/cumulus.git', branch = 'polkadot-v0.9.22' }
+statemint-runtime                     = { git = 'https://github.com/paritytech/cumulus.git', branch = 'polkadot-v0.9.22' }
+xcm-emulator                          = { git = 'https://github.com/shaunxw/xcm-simulator.git', rev = '5a04d6286ee9082d4b7433541e48508f8b65616b' }
```
[the remark this answers is not in the document]
Comment (yrong) [reply integration-tests/Cargo.toml:86]: same,any way it is commit id not secret

**Comment by yrong** [comment]
polkadot customized in this branch
https://github.com/parallel-finance/polkadot/tree/release-v0.9.22-fast-sudo

**Comment by yrong** [comment]
ci for coverage breaks in https://github.com/parallel-finance/parallel/runs/6668255722?check_suite_focus=true
complain about compile failure but try in local setup execution succeed
`cargo +nightly-2022-04-24 tarpaulin --verbose --no-fail-fast -p pallet-liquid-staking --timeout 300 --out Xml`
@GopherJ  have you met this issue before, anything to do with github hosted runners?

**Comment by GopherJ** [comment]
> ci for coverage breaks in https://github.com/parallel-finance/parallel/runs/6668255722?check_suite_focus=true complain about compile failure but try in local setup execution succeed
> 
> `cargo +nightly-2022-04-24 tarpaulin --verbose --no-fail-fast -p pallet-liquid-staking --timeout 300 --out Xml`
> 
> @GopherJ have you met this issue before, anything to do with github hosted runners?

I haven't had this issue before, maybe coverage tool will need to be upgraded.

**Review by GopherJ — APPROVED** [review]

File: Cargo.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (GopherJ) [reply Cargo.toml:17]: yes but I didn't find how to configure reviewdog on this

**Pull request merged**